camping.py

```diff
@@ -266,76 +281,15 @@ def main(parks, json_output=False):
         return output_human_output(parks)
 
 
-def valid_date(s):
-    try:
-        return datetime.strptime(s, INPUT_DATE_FORMAT)
-    except ValueError:
-        msg = "Not a valid date: '{0}'.".format(s)
-        raise argparse.ArgumentTypeError(msg)
-
-
-def positive_int(i):
-    i = int(i)
-    if i <= 0:
-        msg = "Not a valid number of nights: {0}".format(i)
-        raise argparse.ArgumentTypeError(msg)
-    return i
-
-
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--debug", "-d", action="store_true", help="Debug log level")
-    parser.add_argument(
-        "--start-date", required=True, help="Start date [YYYY-MM-DD]", type=valid_date
-    )
-    parser.add_argument(
-        "--end-date",
-        required=True,
-        help="End date [YYYY-MM-DD]. You expect to leave this day, not stay the night.",
-        type=valid_date,
-    )
-    parser.add_argument(
-        "--nights",
-        help="Number of consecutive nights (default is all nights in the given range).",
-        type=positive_int,
-    )
-    parser.add_argument(
-        "--campsite-type",
-        help=(
-            "If you want to filter by a type of campsite. For example "
-            '"STANDARD NONELECTRIC" or TODO'
-        ),
-    )
-    parser.add_argument(
-        "--json-output",
-        action="store_true",
-        help=(
-            "This make the script output JSON instead of human readable "
-            "output. Note, this is incompatible with the twitter notifier. "
-            "This output includes more precise information, such as the exact "
-            "avaiable dates and which sites are available."
-        ),
-    )
-    parks_group = parser.add_mutually_exclusive_group(required=True)
-    parks_group.add_argument(
-        "--parks", dest="parks", metavar="park", nargs="+", help="Park ID(s)", type=int
-    )
-    parks_group.add_argument(
-        "--stdin",
-        "-",
-        action="store_true",
-        help="Read list of park ID(s) from stdin instead",
-    )
-
+    parser = CampingArgParser()
     args = parser.parse_args()
 
     if args.debug:
         LOG.setLevel(logging.DEBUG)
 
-    parks = args.parks or [p.strip() for p in sys.stdin]
```

Comment:
This is handled in CampingArgParser. It does all the validation and argument manipulation when the args are parsed.